openshift_resources/oadp/data_protection_application.py

```diff
@@ -1,17 +1,391 @@
+        description='name is the name requested for the bucket (aws, gcp) or container (azure)',
+    )
+    provider: Provider = Field(
+        ..., description='provider is the provider of the cloud storage'
+    )
+    region: Optional[str] = Field(
+        None,
+        description='region` for the bucket to be in, will be us-east-1 if not set.',
+    )
+    tags: Optional[Dict[str, str]] = Field(None, description='tags` for the bucket')
 
-    def wait_for_failed_reconciliation(self):
-        self.wait_for_condition(condition=r.Condition.CONDITION_RECONCILED.value, status=r.Status.FAILED_TO_RECONCILE.value)
 
+class CloudStorageStatus(BaseModel):
+    lastSyncTimestamp: Optional[datetime] = None
+    name: str
+
+
+class CloudStorageRef(BaseModel):
+    name: Optional[str] = Field(
+        None,
+        description='Name of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names TODO: Add other useful fields. apiVersion, kind, uid?',
+    )
+
+
+class Bucket(BaseModel):
+    backupSyncPeriod: Optional[Any] = Field(
+        None,
+        description='backupSyncPeriod defines how frequently to sync backup API objects from object storage. A value of 0 disables sync.',
+    )
+    cloudStorageRef: CloudStorageRef = Field(
+        ...,
+        description='LocalObjectReference contains enough information to let you locate the referenced object inside the same namespace.',
+    )
+    config: Optional[Dict[str, str]] = Field(
+        None, description='config is for provider-specific configuration fields.'
+    )
+    credential: Optional[SecretKeySelector] = Field(
+        None,
+        description='credential contains the credential information intended to be used with this location',
+    )
+    default: Optional[bool] = Field(
+        None,
+        description='default indicates this location is the default backup storage location.',
+    )
+
+
+class VeleroCredential(BaseModel):
+    key: str = Field(
+        ...,
+        description='The key of the secret to select from.  Must be a valid secret key.',
+    )
+    name: Optional[str] = Field(
+        None,
+        description='Name of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names TODO: Add other useful fields. apiVersion, kind, uid?',
+    )
+    optional: Optional[bool] = Field(
+        None, description='Specify whether the Secret or its key must be defined'
+    )
+
+
+class ObjectStorage(BaseModel):
+    bucket: str = Field(
+        ..., description='Bucket is the bucket to use for object storage.'
+    )
+    caCert: Optional[str] = Field(
+        None,
+        description='CACert defines a CA bundle to use when verifying TLS connections to the provider.',
+    )
+    prefix: Optional[str] = Field(
+        None,
+        description='Prefix is the path inside a bucket to use for Velero storage. Optional.',
+    )
+
+
+class VeleroConfig(BaseModel):
```

Comment:
should be called CloudStorageLocation.
 See:
https://pkg.go.dev/github.com/openshift/oadp-operator/api/v1alpha1#CloudStorageLocation